### PR TITLE
Only add internal files if the internal metadata path exists

### PR DESCRIPTION
**Pull request by crobibero**
Fixes regression from https://github.com/jellyfin/jellyfin/pull/7255/

### CODE REVIEW

File: MediaBrowser.Providers/MediaInfo/MediaInfoResolver.cs [hunk 1]
```diff
@@ -148,15 +148,19 @@ public abstract class MediaInfoResolver
...
             }
 
             var externalPathInfos = new List<ExternalPathParserResult>();
 
             var files = directoryService.GetFilePaths(folder, clearCache).ToList();
-            files.AddRange(directoryService.GetFilePaths(video.GetInternalMetadataPath(), clearCache));
+            var internalMetadataPath = video.GetInternalMetadataPath();
+            if (directoryService.PathExists(internalMetadataPath))
+            {
+                files.AddRange(directoryService.GetFilePaths(video.GetInternalMetadataPath(), clearCache));
```
Comment (1337joe) [inline MediaBrowser.Providers/MediaInfo/MediaInfoResolver.cs:162]:
```suggestion
                files.AddRange(directoryService.GetFilePaths(internalMetadataPath, clearCache));
```

**Review by 1337joe — APPROVED** [review]

**Comment by 1337joe** [comment]
The changes to the tests now mock that all directories exist for `GetExternalStreams_BadPaths_ReturnsNoSubtitles` where part of the intent of that was to test a non-existing path.

Here's an update that covers the state you saw fail (missing metadata directory) and improves the verification of path not found to check the actual calls to the mock instead of the results: https://github.com/1337joe/jellyfin/commit/73d3aaaa8b543838c1507e824ad3b3c2e1ad520c

Feel free to pull it into this pull request or I'll make a new pull request after this merges.

**Review by 1337joe — DISMISSED** [review]

File: MediaBrowser.Providers/MediaInfo/MediaInfoResolver.cs [hunk 2]
```diff
@@ -148,15 +148,19 @@ public abstract class MediaInfoResolver
 
             // Check if video folder exists
             string folder = video.ContainingFolderPath;
-            if (!Directory.Exists(folder))
+            if (!directoryService.PathExists(folder))
             {
                 return Array.Empty<ExternalPathParserResult>();
             }
 
             var externalPathInfos = new List<ExternalPathParserResult>();
```
Comment (cvium) [inline MediaBrowser.Providers/MediaInfo/MediaInfoResolver.cs:156]: Not related to your PR, but it's a bit awkward that there's a new List here when we return an empty array just a few lines down

File: MediaBrowser.Controller/Providers/IDirectoryService.cs [hunk 3]
```diff
@@ -16,5 +16,12 @@ public interface IDirectoryService
         IReadOnlyList<string> GetFilePaths(string path);
 
         IReadOnlyList<string> GetFilePaths(string path, bool clearCache, bool sort = false);
+
+        /// <summary>
+        /// Does the path exist.
+        /// </summary>
+        /// <param name="path">The path.</param>
+        /// <returns>Whether the path exists.</returns>
+        bool PathExists(string path);
```
Comment (cvium) [inline MediaBrowser.Controller/Providers/IDirectoryService.cs:25]: Should we not simply call it `Exists`? I'm not at all keen on the `IDirectoryService` interface, but I see the benefits when it comes to testing.

**Comment by crobibero** [comment]
Test not passing: GetExternalFiles_MissingDirectory_DirectoryNotQueried

File: MediaBrowser.Controller/Providers/DirectoryService.cs [hunk 4]
```diff
@@ -3,6 +3,7 @@
 using System;
 using System.Collections.Concurrent;
 using System.Collections.Generic;
+using System.IO;
```
Comment (1337joe) [inline MediaBrowser.Controller/Providers/DirectoryService.cs:6]: Import not used.
```suggestion
```

**Review by 1337joe — APPROVED** [review]

File: Emby.Server.Implementations/IO/ManagedFileSystem.cs [hunk 5]
```diff
@@ -704,6 +704,12 @@ public virtual IEnumerable<string> GetFileSystemEntryPaths(string path, bool rec
             return Directory.EnumerateFileSystemEntries(path, "*", GetEnumerationOptions(recursive));
         }
 
+        /// <inheritdoc />
+        public virtual bool Exists(string path)
+        {
+            return Directory.Exists(path) || File.Exists(path);
```
Comment (Bond-009) [inline Emby.Server.Implementations/IO/ManagedFileSystem.cs:710]: Only the `Directory.Exists(path)` path get used now, no?
Shouldn't we split this up into 2 functions, `DirectoryExists` and `FileExists` or something.
The current code will do 2 syscalls if it the dir doesn't exist or if it's a file (I think)

**Review by Bond-009 — APPROVED** [review]

**Review by cvium — APPROVED** [review]

**Pull request merged**